### PR TITLE
add tests for our fetch wrapper and federalistApi

### PR DESCRIPTION
**Pull request by jseppi**
Adds tests for our front end's `fetch` wrapper code and for `federalistApi`. This code was previously untested.

ref #1132.

Was going to try to get this into #1157 but @jmhooper got to the review just moments before I finished this PR :)

### CODE REVIEW

**Comment by jseppi** [comment]
`+3.2%` front end coverage 🎉

File: test/frontend/util/datetime.test.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ const DURATION = 'a few seconds';
 const NO_DURATION = '-';
 
 
-describe('datetime', function() {
+describe('datetime', () => {
```
Comment (jseppi) [inline test/frontend/util/datetime.test.js]: just linting fixes here

File: test/frontend/util/fetch.test.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+global.fetch = fetchMock;
+
+describe('fetchWrapper', () => {
+  before(() => {
+    fetchMock.put('/horses', {}, { name: 'putHorses' });
+    fetchMock.post('/cats', {}, { name: 'postCats' });
+    fetchMock.get('/cats', JSON.stringify({ cats: 5 }), { name: 'cats' });
+    fetchMock.get('/dogs', { say: 'woof' }, { name: 'dogs' });
+    fetchMock.get('/dogs/bad', { status: 400, body: 'they\'re all good dogs, brent' }, { name: 'badDogs' });
```
Comment (jseppi) [inline test/frontend/util/fetch.test.js]: 🐶

File: test/frontend/util/fetch.test.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+chai.use(chaiFetchMock);
+
+global.fetch = fetchMock;
+
+describe('fetchWrapper', () => {
+  before(() => {
+    fetchMock.put('/horses', {}, { name: 'putHorses' });
+    fetchMock.post('/cats', {}, { name: 'postCats' });
+    fetchMock.get('/cats', JSON.stringify({ cats: 5 }), { name: 'cats' });
```
Comment (jseppi) [inline test/frontend/util/fetch.test.js]: 🐱

File: test/frontend/util/fetch.test.js [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+    fetchMock.get('/dogs', { say: 'woof' }, { name: 'dogs' });
+    fetchMock.get('/dogs/bad', { status: 400, body: 'they\'re all good dogs, brent' }, { name: 'badDogs' });
+    fetchMock.get('/cats/affection', { status: 204 }, { name: 'affectionCats' });
```
Comment (jseppi) [inline test/frontend/util/fetch.test.js]: `404` would have been more clever, but I wanted to test the `204` behavior

**Comment by jseppi** [comment]
Actually, @jmhooper, if you can wait a little while I'll add a few `federalistApi` tests to this PR too. It would make me more comfortable to at least have some smoke tests.

**Comment by jseppi** [comment]
Alright -- got in some tests for `federalistApi` now 💥

**Comment by jseppi** [comment]
Front end coverage now up by `+5.3%` because of this PR 🌯 🦅

File: test/frontend/util/federalistApi.test.js [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+    expectedOptions.body = JSON.stringify(body);
+  }
+
+  expect(fetchMock).route(routeName).to.have.been.called
+    .with.options(expectedOptions);
+}
+
+describe('federalistApi', () => {
+  before(() => {
+    fetchMock.get(`${API}/site/${testSite.id}/build`, { build: true }, { name: 'getBuilds' });
```
Comment (jmhooper) [inline test/frontend/util/federalistApi.test.js]: Nice, this is much cleaner than I was anticipating

File: frontend/util/federalistApi.js [hunk 6]
```diff
@@ -5,15 +5,15 @@ import alertActions from '../actions/alertActions';
...
-const csrfToken = typeof window !== 'undefined' ?
-  window.CSRF_TOKEN : global.CSRF_TOKEN;
-
-const defaultHeaders = {
-  'x-csrf-token': csrfToken,
-};
-
 export default {
   fetch(endpoint, params = {}) {
+    const csrfToken = typeof window !== 'undefined' ?
```
Comment (jmhooper) [inline frontend/util/federalistApi.js]: Was this move to get the tests to behave after you require `federalistAPI`? Frustrating, but 🤷‍♀️, I don't have a better way to manage global state.

Comment (jseppi) [reply frontend/util/federalistApi.js]: yep, exactly

**Pull request merged**